File: RxNetWorkLibrary/src/main/java/io/reactivex/network/manager/RxNetWorkListener.java
package io.reactivex.network.manager;

import android.support.annotation.NonNull;

import io.reactivex.Observable;
import retrofit2.Retrofit;

/**
 * by y on 2017/2/22.
 */

public interface RxNetWorkListener<T> {

    void onNetWorkStart();

    void onNetWorkError(Throwable e);

    void onNetWorkComplete();

    void onNetWorkSuccess(T data);

    @NonNull
    Observable<T> getObservable(Retrofit retrofit);
}

File: RxNetWorkLibrary/src/main/java/io/reactivex/network/RxNetWorkListener.kt
package io.reactivex.network

/**
 * by y on 2017/2/22.
 */

interface RxNetWorkListener<in T> {

    fun onNetWorkStart()

    fun onNetWorkError(e: Throwable)

    fun onNetWorkComplete()

    fun onNetWorkSuccess(data: T)

}
